SpeechResponder/ConfigurationWindow.xaml.cs

```diff
@@ -7,379 +7,404 @@
-        {
-            InitializeComponent();
-            DataContext = this;
+		public ConfigurationWindow()
+		{
+			InitializeComponent();
+			DataContext = this;
 
-            ObservableCollection<Personality> personalities = new ObservableCollection<Personality>
-            {
-                // Add our default personality
-                Personality.Default()
-            };
-            // Add local personalities
-            foreach (Personality personality in Personality.AllFromDirectory())
-            {
-                if (personality != null)
-                {
-                    personalities.Add(personality);
-                }
-            }
-            Personalities = personalities;
+			ObservableCollection<Personality> personalities = new ObservableCollection<Personality>
+			{
+				// Add our default personality
+				Personality.Default()
+			};
+			// Add local personalities
+			foreach (Personality personality in Personality.AllFromDirectory())
+			{
+				if (personality != null)
+				{
+					personalities.Add(personality);
+				}
+			}
+			Personalities = personalities;
 
-            SpeechResponderConfiguration configuration = SpeechResponderConfiguration.FromFile();
-            subtitlesCheckbox.IsChecked = configuration.Subtitles;
-            subtitlesOnlyCheckbox.IsChecked = configuration.SubtitlesOnly;
+			SpeechResponderConfiguration configuration = SpeechResponderConfiguration.FromFile();
+			subtitlesCheckbox.IsChecked = configuration.Subtitles;
+			subtitlesOnlyCheckbox.IsChecked = configuration.SubtitlesOnly;
 
-            foreach (Personality personality in Personalities)
-            {
-                if (personality.Name == configuration.Personality)
-                {
-                    Personality = personality;
-                    break;
-                }
-            }
-        }
+			foreach (Personality personality in Personalities)
+			{
+				if (personality.Name == configuration.Personality)
+				{
+					Personality = personality;
+					break;
+				}
+			}
 
-        private void eddiScriptsEnabledUpdated(object sender, RoutedEventArgs e)
-        {
-            if (sender is CheckBox checkbox)
-            {
-                if (checkbox.IsLoaded)
-                {
-                    updateScriptsConfiguration();
-                }
-            }
-        }
+			Dispatcher.BeginInvoke(new Action(() =>
+			{
+				var recoveredScript = ScriptRecoveryService.GetRecoveredScript();
+				if (recoveredScript != null)
+				{
+					var messageBoxResult = MessageBox.Show(Properties.SpeechResponder.messagebox_recoveredScript,
+						Properties.SpeechResponder.messagebox_recoveredScript_title,
+						MessageBoxButton.YesNo, MessageBoxImage.Warning, MessageBoxResult.Yes,
+						MessageBoxOptions.DefaultDesktopOnly);
+					if (messageBoxResult == MessageBoxResult.Yes)
+					{
+						OpenEditScriptWindow(recoveredScript);
+					}
+				}
+			}), DispatcherPriority.ApplicationIdle);
+		}
```

Comment:
Depending on when the crash occurs, the UI may not save the most recent tab state and may load a prior selected tab. Should we trigger this when the Speech responder tab is selected in the UI rather than when the MainWindow and child tabs are first loaded?

Reply:
That is no problem at all. The Constructor of the ConfigurationWindow is invoked even when the tab is not visible. If the application has crashed, the last thing the user was doing, was editing a script, so it is expected that the user will restart the application to continue to edit the script.

I thought of moving the recovery step to a more central place but decided that for the separation of concerns thats fine.

Reply:
Fair enough. I suppose if the tab state isn't getting saved often enough then that's a separate concern. ;-)